modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -212,73 +246,89 @@ public CoreEvent process(CoreEvent event) throws MuleException {
+                    } catch (MuleException t) {
+                      throw propagateWrappingFatal(localOperatorErrorHook.apply(t, event));
+                    }
+                  })
+                  .doOnNext(event -> {
+                    final ExecutorCallback executorCallback = new ExecutorCallback() {
+
+                      @Override
+                      public void error(Throwable e) {
+                        // if `sink.error` is called here, it will cancel the flux altogether.
+                        // That's why an `Either` is used here,
+                        // so the error can be propagated afterwards in a way consistent with our expected error handling.
+                        errorSwitchSinkSinkRef.next(left(
+                                                         // Force the error mapper from the chain to be used.
+                                                         // When using Mono.create with sink.error, the error mapper from the
+                                                         // context is ignored, so it has to be explicitly used here.
+                                                         localOperatorErrorHook.apply(e, event), CoreEvent.class));
+                      }
+
+                      @Override
+                      public void complete(Object value) {
+                        errorSwitchSinkSinkRef.next(right(Throwable.class, (CoreEvent) value));
+                      }
+                    };
+
+                    if (!isAsync() && NO_POLICY_OPERATION
+                        .equals(((SdkInternalContext) ((InternalEvent) event).<SdkInternalContext>getSdkInternalContext())
+                            .getPolicyToApply())) {
+                      onEventSynchronous(event, executorCallback, ctx);
+                    } else {
+                      onEvent(event, executorCallback);
+                    }
+                  }), () -> errorSwitchSinkSinkRef.complete(), t -> errorSwitchSinkSinkRef.error(t),
+                                                                 muleContext.getConfiguration().getShutdownTimeout(),
+                                                                 outerFluxCompletionScheduler));
+
+          if (publisher instanceof Flux && !ctx.getOrEmpty(WITHIN_PROCESS_TO_APPLY).isPresent()) {
+            return transformed
+                .doAfterTerminate(this::outerPublisherTerminated)
+                .doOnSubscribe(s -> outerPublisherSubscribedTo());
+          } else {
+            // Certain features (ext client, batch, flow runner) use Mono, so we don't want to dispose the inner stuff after the
+            // first event comes through
+            return transformed;
+          }
+        });
   }
 
   private void onEvent(CoreEvent event, ExecutorCallback executorCallback) {
     try {
+      SdkInternalContext sdkInternalContext =
+          ((SdkInternalContext) ((InternalEvent) event).<SdkInternalContext>getSdkInternalContext());
```

Comment:
extract to static method in `org.mule.runtime.module.extension.api.util.MuleExtensionUtils`